scripts/cleanup-test-env.sh

```diff
@@ -16,10 +16,33 @@
 
 set -Eeou pipefail
 
+delete_endpoint() {
+    provider=$1
+    set +e
+    count=$(atlas privateEndpoints "${provider}" list --projectId "${clean_project_id}" -o=go-template="{{len .}}")
+    retVal=$?
+    if [ $retVal -ne 0 ]; then
+        count=0
+    fi
+    if [ "${count}" != "0" ]; then
+        echo "Project ${clean_project_id} contains ${provider} endpoints, will start deleting it now and will try to delete the project in the next execution"
+        id=$(atlas privateEndpoints "${provider}" list --projectId "${clean_project_id}" -o=go-template="{{(index . 0).Id}}")
```

Comment:
the list commands return the provider as well
```json
[
  {
    "cloudProvider": "AWS",
    "id": "64c90604ca62bc79f4554b5c",
    "regionName": "EU_WEST_1",
    "status": "AVAILABLE",
    "endpointServiceName": "com.amazonaws.vpce.eu-west-1.vpce-svc-032aa573f24af944c",
    "interfaceEndpoints": [
      "vpce-07d624b0203224579"
    ]
  }
]
```

instead of calling the function 3 times, could you just use the provider in the response as you are doing for the ID?

Reply:
I'm not sure I can do that the provider is part of the command, the CLI cleanup also calls 3 times.

e.g.:  atlas privateEndpoints azure list --projectId my_project_id

I don't think you can ask for all endpoints in one call